VSDiagnostics/VSDiagnostics/VSDiagnostics/Diagnostics/General/EqualsAndGetHashcodeNotImplementedTogether/EqualsAndGetHashcodeNotImplementedTogetherAnalyzer.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System.Collections.Generic;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using VSDiagnostics.Utilities;
+
+namespace VSDiagnostics.Diagnostics.General.EqualsAndGetHashcodeNotImplementedTogether
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class EqualsAndGetHashcodeNotImplementedTogetherAnalyzer : DiagnosticAnalyzer
+    {
+        private const DiagnosticSeverity Severity = DiagnosticSeverity.Warning;
+
+        private static readonly string Category = VSDiagnosticsResources.GeneralCategory;
+        private static readonly string Message = VSDiagnosticsResources.EqualsAndGetHashcodeNotImplementedTogetherAnalyzerMessage;
+        private static readonly string Title = VSDiagnosticsResources.EqualsAndGetHashcodeNotImplementedTogetherAnalyzerTitle;
+
+        internal static DiagnosticDescriptor Rule
+            => new DiagnosticDescriptor(DiagnosticId.EqualsAndGetHashcodeNotImplementedTogether, Title, Message, Category, Severity, true);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context) => context.RegisterSyntaxNodeAction(AnalyzeNode, SyntaxKind.ClassDeclaration);
+
+        private void AnalyzeNode(SyntaxNodeAnalysisContext context)
+        {
+            var classDeclaration = (ClassDeclarationSyntax)context.Node;
+
+            var equalsImplemented = false;
+            var getHashcodeImplemented = false;
+
+            foreach (var node in classDeclaration.Members)
+            {
+                if (!node.IsKind(SyntaxKind.MethodDeclaration))
+                {
+                    continue;
+                }
+
+                var methodDeclaration = (MethodDeclarationSyntax)node;
+
+                if (!IsOverride(methodDeclaration))
+                {
+                    continue;
+                }
+
+                if (methodDeclaration.Identifier.ValueText == nameof(Equals) && methodDeclaration.ParameterList.Parameters.Count == 1)
```

Comment:
I don't like this approach. However there is a more solid alternative: you can request the symbol of `Object+Equals(object)` with something like this:

	var objectSymbol = compilation.GetSpecialType(SpecialType.System_Object);
	var objectEquals = objectSymbol.GetMembers().OfType<IMethodSymbol>().Where(x => x.MetadataName == "Equals" && x.Parameters.Count() == 1);

Afterwards you can use `context.SemanticModel.GetDeclaredSymbol(methodDeclaration).OverriddenMethod` and compare it to the the `Object+Equals(object)` method. 

Some things to note:

* Fetch `Object+Equals(object)` only once per compilation. You can do this by registering a compilation action in `Initialize()`
* You have to iterate through the inheritance structure in case a parent class already overrides it